Lightyear/Core.idr

```diff
@@ -153,4 +153,13 @@ satisfy : (Monad m, Stream tok str)
                    => (tok -> Bool)
                    -> ParserT str m tok
 satisfy p = satisfyMaybe (\t => if p t then Just t else Nothing)
+
+||| Succeed if and only if the argument parser fails.
+notFollowedBy : ParserT str m tok -> ParserT str m ()
```

Comment:
I'm thinking about keeping `notFollowedBy` as a deprecated alias for `expectFailure` to stay friendly to Parsec users. From the alternatives so far, I like `expectFailure` most because it says when the parser succeeds, instead of saying when the parser fails. I find descriptions by negation unclear because of reliance on the closed-world assumption.